### PR TITLE
wip

### PR DESCRIPTION
**Pull request by seallard**
Split rules based on which model they operate on instead of the type of rule.

### CODE REVIEW

File: cg/services/order_validation_service/model_validator/model_validator.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from cg.services.order_validation_service.model_validator.utils import convert_errors
+from cg.services.order_validation_service.models.order import Order
+
+
+class ModelValidator:
+
+    @staticmethod
+    def validate(order: dict, model: Order) -> tuple[Order | None, ValidationErrors]:
+        try:
+            order = model.model_validate(order)
```
[the remark this answers is not in the document]
Comment (seallard) [reply cg/services/order_validation_service/model_validator/model_validator.py:12]: No, it is just the class itself. How can we convey that more clearly?

[the remark this answers is not in the document]
Comment (islean) [reply cg/services/order_validation_service/model_validator/model_validator.py:12]: Should it not be a workflow which we can map to the correct class. and then use the class method `model_validate` on?

File: cg/services/order_validation_service/rules/case_sample/utils.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+def is_concentration_missing(sample: TomteSample) -> bool:
+    return not sample.concentration_ng_ul
+
+
+def is_well_position_missing(sample: TomteSample) -> bool:
+    return sample.container == ContainerEnum.plate and not sample.well_position
+
+
+def is_container_name_missing(sample: TomteSample) -> bool:
```
Comment (islean) [inline cg/services/order_validation_service/rules/case_sample/utils.py:27]: I am a bit uncertain if this is a case_sample util or a sample util. Same goes for a lot of these I guess. Regardless of workflow I think you can specify container, definitely for some standalone sample workflows

File: cg/services/order_validation_service/model_validator/model_validator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (seallard) [reply cg/services/order_validation_service/model_validator/model_validator.py:12]: I think it is more straightforward to provide the class instead, it makes the class self contained. What about adding a type hint conveying that it is not an instance, like this
```python
validate(order: dict, model: Type[Order]) -> tuple[Order | None, ValidationErrors]
```

File: cg/services/order_validation_service/rules/order/rules.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    **kwargs,
+) -> list[CustomerCannotSkipReceptionControlError]:
+    errors: list[CustomerCannotSkipReceptionControlError] = []
+
+    if not order.skip_reception_control:
+        return errors
+
+    if not store.is_customer_trusted(order.customer):
+        error = CustomerCannotSkipReceptionControlError()
+        errors.append(error)
```
Comment (islean) [inline cg/services/order_validation_service/rules/order/rules.py:73]: For some reason I am not allowed to comment on this file in PyCharm but could we do,
```suggestion

    if order.skip_reception_control and not store.is_customer_trusted(order.customer):
        error = CustomerCannotSkipReceptionControlError()
        errors.append(error)
```

File: cg/services/order_validation_service/model_validator/model_validator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (islean) [reply cg/services/order_validation_service/model_validator/model_validator.py:12]: Or maybe `Type[Order]`

File: cg/services/order_validation_service/rules/case_sample/utils.py (hunk 2 above)
Comment (seallard) [reply cg/services/order_validation_service/rules/case_sample/utils.py:27]: Currently, they are producing `CaseSampleErrors` and as such, they belong in the `case_sample` module. They also operate on Tomte specific types, which means that they belong in the tomte module. I'll move them there.

File: cg/services/order_validation_service/model_validator/model_validator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (islean) [reply cg/services/order_validation_service/model_validator/model_validator.py:12]: Do we not lose something when the returned Order is just the general parent class though? Or maybe that is fine?

File: cg/services/order_validation_service/rules/case_sample/utils.py (hunk 2 above)
Comment (islean) [reply cg/services/order_validation_service/rules/case_sample/utils.py:27]: That is not what I am after. This bool will need to be calculated for all workflows. As such the utils function (not the whole rule) could very well be placed in a different dir.

File: cg/services/order_validation_service/model_validator/model_validator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (seallard) [reply cg/services/order_validation_service/model_validator/model_validator.py]: That is the same thing. https://github.com/Clinical-Genomics/cg/pull/3623/commits/3068478f2ca3ffb53e4f5a6f309904238df6d7b7

[the remark this answers is not in the document]
Comment (islean) [reply cg/services/order_validation_service/model_validator/model_validator.py]: I do not understand what you mean. Type hinting the returned type as `TomteOrder` is not the same as typing it as `Order`.

[the remark this answers is not in the document]
Comment (seallard) [reply cg/services/order_validation_service/model_validator/model_validator.py]: Oh, I did not see you second comment. This is a general class now and should not mention any specific sub-classes of the `Order`. The correct type hints will need to be used where ever the class is used

File: cg/services/order_validation_service/rules/case_sample/utils.py (hunk 2 above)
Comment (seallard) [reply cg/services/order_validation_service/rules/case_sample/utils.py:27]: I'll leave this for separate PR:s, as discussed in slack

File: cg/services/order_validation_service/rules/order/rules.py (hunk 3 above)
Comment (seallard) [reply cg/services/order_validation_service/rules/order/rules.py]: https://github.com/Clinical-Genomics/cg/pull/3623/commits/e90f2c9e490d214d7ae4f66d2c4af8178d37d1d1

**Pull request merged**